### PR TITLE
Record table column changes

### PR DESCRIPTION
**Pull request by ranisa-gupta16**
## JIRA Ticket

[BSS-447](https://jira.csiro.au/browse/BSS-447)

## Description

This PR is related to layout changes. enhancements made in "Record Table" , and 'Data Grid Toolbar ' components with improved table layout , design and styling, and improved table search bar, clear button embeeded in search bar, and filter button aligned in same row as search bar.

## Proposed Changes

Search Bar and Filter Enhancements:

Improved search bar functionality with clearer placeholder text.
Added distinct hover, focus, and clear button interactions.
Enhanced search filter functionality for better user experience.
Optimized performance of the clear button to reset filters quickly.
Layout and Styling Updates for Record Table:

Improved table layout for better readability and usability.
Made the table more responsive, adapting automatically based on screen size.
Enhanced styling for table rows, headers, and cell contents to make data more prominent and visible.
Added clearer distinctions between different column types for better visual hierarchy.
Responsiveness:

Removed the mobile toggle switch, as the table now adjusts responsively based on screen size.
Ensured that both desktop and mobile views offer a clean, optimized experience for users.
Code Documentation and Maintainability:

Added JSDoc-style comments to the RecordTable component for better code readability and easier maintenance.
Updated inline comments throughout the layout and table structure for clearer understanding of the code.
Lint Fixes:

Addressed TypeScript linting errors, particularly related to inferred types for booleans, and removed redundant type annotations.
Ensured all files comply with the project's linting and coding standards.

__Steve Added__: Updated the RecordTable to look at visible_types and summary_fields:

- only include records who's type is in visible_types in the record list (Sites, not Buildings)
- if there is only one visible_type AND there are configured summary_fields, show those fields instead of the HRID
- Simplify the default HRID column display, don't use the term HRID and remove some clutter
- Make the columns included in the table scale with the page width, add more columns as it gets bigger
- Verified that it works ok for Rapid Site Survey and other notebooks with more types

## How to Test

Please View Records table of a Survey/Notebook.   

## Checklist

- [x] I have confirmed all commits have been signed.
- [x] I have added JSDoc style comments to any new functions or classes.
- [x] Relevant documentation such as READMEs, guides, and class comments are updated.

### CODE REVIEW

**Comment by stevecassidy** [comment]
The 'Filters' button is much improved but seems to be just off the edge of the screen in many display widths. This looks like a good basis to improve the column layout of the notebook list, thanks for splitting it out.

**Comment by ranisa-gupta16** [comment]
> The 'Filters' button is much improved but seems to be just off the edge of the screen in many display widths. This looks like a good basis to improve the column layout of the notebook list, thanks for splitting it out.

Yes, I did a testing on different widths and found same cutting off around the edge issue. Will update some styling on this

**Comment by ranisa-gupta16** [comment]
Adding a note:  This branch will be used to address issues with Records listings currently under discussion.

File: app/src/gui/components/notebook/record_table.tsx [hunk 1]
```diff
@@ -62,29 +81,53 @@ type RecordsBrowseTableProps = {
...
+  useEffect(() => {
+    const get = async () => {
+      const ui = await getUiSpecForProject(project_id);
+      setUISpec(ui);
+      const visible = getVisibleTypes(ui);
+      setVisibleTypes(visible);
+    };
+
+    get();
+  }, [props.project_id]);
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/record_table.tsx:115]: This should be using custom hook / react query - but okay for now

File: app/src/gui/components/notebook/record_table.tsx [hunk 2]
```diff
@@ -107,283 +156,238 @@ function RecordsTable(props: RecordsTableProps) {
...
-              <RecordDelete
-                project_id={project_id}
-                record_id={params.row.record_id}
-                revision_id={params.row.revision_id}
-                draft_id={null}
-                show_label={false}
-                handleRefresh={props.handleRefresh}
-              />
+              <Typography>
+                {params.row.data ? params.row.data[field] : 'missing'}
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/record_table.tsx:309]: Is this a likely condition? If so maybe need more thought than 'missing'?

File: app/src/gui/components/notebook/record_table.tsx
Comment (PeterBaker0) [inline app/src/gui/components/notebook/record_table.tsx:1]: how does the component remount on layout change to update the columns?

**Review by luke-mcfarlane-rocketlab — APPROVED** [review]
LGTM

**Review by PeterBaker0 — APPROVED** [review]
Some minor preferences regarding column headers etc but good to merge for this demo. Good work.

**Pull request merged**